forc-pkg/src/pkg.rs

```diff
@@ -1609,9 +1610,17 @@ where
 {
     // Clear existing temporary directory if it exists.
     let repo_dir = tmp_git_repo_dir(fetch_id, name, &source.repo);
-    if repo_dir.exists() {
-        let _ = std::fs::remove_dir_all(&repo_dir);
-    }
+
+    let _ = std::fs::create_dir_all(&repo_dir);
+    let mut lock = RwLock::new(
+        fs::OpenOptions::new()
+            .write(true)
+            .create(true)
+            .open(repo_dir.join(FORC_FILE_LOCK_NAME))?,
+    );
+    let _ = lock.write()?;
+
+    let _ = std::fs::remove_dir_all(&repo_dir);
```

Comment:
Wait, if we remove the directory here, don't we also remove our lock file?

Reply:
I should probably convert this to a draft - I've realized now that in my original PR it was incorrectly implemented as well. Also, `file-lock` is based on `fnctl()` which is different from `fd-lock` which is based on `flock()`. For whatever reason initializing the lock at the same line did not work for `fd-lock` - still looking into this and trying to understand how flock() compares against fnctl().